src/crypto/sodium.cpp

```diff
@@ -107,52 +113,141 @@ int EdDSA::verify(const unsigned char* msg, unsigned long long msglen,
                                   signedmsg, msglen + crypto_sign_BYTES, pubKey);
     free(msgbuffer);
     free(signedmsg);
-    if (result) {
-        return(1);
-    } else {
-        // crypto_sign_open() returns -1 on failure.
-        return(0);
-    }
+
+    // crypto_sign_open() returns 0 on success.
+    return result ? 1 : 0;
 }
 
 
 // Generates a new Ed25519 private key seed. The key seed is stored in the object.
-int EdDSA::genKeySeed(unsigned char* privKey) {
+int EdDSA::genKeySeed(unsigned char* privKey)
+{
     // Make space for a new key seed (if not present).
-    if (!this->keySeed) {
+    if (!this->keySeed)
+    {
         this->keySeed = (unsigned char*)malloc(crypto_sign_SEEDBYTES);
-        if (this->keySeed == NULL) {
+        if (this->keySeed == NULL)
+        {
             // Something went wrong allocating the memory.
-            return(0);
+            return 0;
         }
     }
+
     // Now make the new key seed.
-    this->rng.GenerateBlock(this->keySeed, crypto_sign_SEEDBYTES);
+    PrnGen::genblock(this->keySeed, crypto_sign_SEEDBYTES);
+
     // Copy it to privKey before returning.
     if (privKey)
     {
         memcpy(privKey, this->keySeed, crypto_sign_SEEDBYTES);
     }
-    return(1);
+
+    return 1;
 }
 
 
 // Derives the Ed25519 public key from the stored private key seed.
-int EdDSA::publicKey(unsigned char* pubKey) {
+int EdDSA::publicKey(unsigned char* pubKey)
```

Comment:
Good idea! Then, in this function, we can check if the private key already exists and, instead of deriving the public key (and private key) from the seed, do it directly from the private key: `crypto_sign_ed25519_sk_to_pk()`